wasm-pack.yaml

```diff
@@ -0,0 +1,158 @@
+      patches: upgrade-deps.patch
+
+  - runs: |
+      # prepare
+      rm -rf Cargo.lock
+      cargo fetch --target="${{host.triplet.rust}}"
+
+      cargo auditable build --release --frozen
+      install -Dm755 target/release/wasm-pack \
+      -t "${{targets.contextdir}}"/usr/bin
+
+  - uses: strip
+
+update:
+  enabled: true
+  github:
+    identifier: rustwasm/wasm-pack
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - nodejs
+        - npm
+        - rustup
+        - yarn
+        - build-base
+        - binaryen
+  pipeline:
+    - runs: |
+        # Function to display messages
+        function print_info {
+            echo -e "\033[1;34m[INFO]\033[0m $1"
+        }
+
+        function print_error {
+            echo -e "\033[1;31m[ERROR]\033[0m $1"
+        }
+
+        # Check if wasm-pack is installed
+        if ! command -v wasm-pack &> /dev/null
+        then
+            print_error "wasm-pack could not be found. Please install it first."
+            exit 1
+        fi
+
+        print_info "Installing default Rust toolchain via Rustup..."
+        rustup install stable || print_error "Failed to install Rust via Rustup"
+        rustup default stable || print_error "Failed to set default Rust toolchain"
+        ARCH=$(uname -m)
+        export PATH="$HOME/.rustup/toolchains/stable-${ARCH}-unknown-linux-gnu/bin:$PATH"
+        cargo version
+
+        # Ensure the wasm32-unknown-unknown target is installed
+        print_info "Checking for wasm32-unknown-unknown target..."
+        set +e
+        rustup target list --installed | grep wasm32-unknown-unknown &> /dev/null
+        if [ $? -ne 0 ]; then
+            print_info "Installing wasm32-unknown-unknown target..."
+            rustup target add wasm32-unknown-unknown || print_error "Failed to install wasm32-unknown-unknown target."
+        else
+            print_info "wasm32-unknown-unknown target is already installed."
+        fi
+        set -e
+
+        print_info "Checking for wasm32-wasi target..."
+        rustup target list
+        # Create a new Rust library project
+        PROJECT_NAME="wasm_pack_test"
+        print_info "Creating a new Rust project: $PROJECT_NAME"
+        cargo new --lib $PROJECT_NAME
+        cd $PROJECT_NAME || { print_error "Failed to enter project directory."; exit 1; }
+
+        mv ../Cargo.toml Cargo.toml
+        # replace $PROJECT_NAME with the actual project name in Cargo.toml
+        sed -i "s/\$PROJECT_NAME/$PROJECT_NAME/g" Cargo.toml
+        mv ../src/lib.rs src/lib.rs
+
+        # Build the project with wasm-pack
+        print_info "Building the project with wasm-pack"
+        rustup target add wasm32-unknown-unknown || print_error "Failed to install wasm32-unknown-unknown target."
+        wasm-pack build --target web || { print_error "Build failed."; exit 1; }
+
+        # Verify the build output
+        print_info "Verifying the build output"
+        if [ -d "pkg" ]; then
```

Comment:
Don't need this `if` and the `else` below, if the dir doesn't exist we'll fail to get the file on the next line anyway.